File: HelpSoftware/Help/Help.EF/Entitys/ArticleTag.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Help.EF
{
  public class ArticleTag : IHelpEntity
  {
    [Key]
    public virtual int ID_ArticleTag { get; set; }

    public virtual Article Article { get; set; }

    public virtual Tag Tag { get; set; }

    public virtual DateTime AssingTime { get; set; }

    public virtual User User { get; set; }

    [Timestamp]
    public virtual byte[ ] TimeStamp { get; set; }

    public string GetEntityName()
    {
      return "ArticleTag";
    }

    public int GetID()
    {
      return ID_ArticleTag;
    }

    public bool GetWriteHistoryEntry()
    {
      return true;
    }
  }
}

File: HelpSoftware/Help/Help.Web/Web/src/app/EF/ArticleTag.ts
import { Article, Tag, User } from "./EF";

export class ArticleTag{
    ID_ArticleTag: Number;
    Article: Article;
    Tag: Tag;
    AssingTime: Date;
    User: User;

   deserialize(input){
    this.ID_ArticleTag = input.ID_ArticleTag;
    this.Article = new Article().deserialize(input.Article);
    this.Tag = new Tag().deserialize(input.Tag);
    this.AssingTime = input.AssingTime;
    this.User = new User().deserialize(input.User);
    return this;
   }
}